rpc/types/src/misc/distribution.rs

```diff
@@ -15,30 +15,39 @@ use cuprate_epee_encoding::{
 };
 
 //---------------------------------------------------------------------------------------------------- Free
-/// Used for [`Distribution::CompressedBinary::compressed_data`].
-///
-/// TODO: for handler code. This should already be provided in the field.
+/// Used for [`Distribution::CompressedBinary::distribution`].
 #[doc = crate::macros::monero_definition_link!(
     cc73fe71162d564ffda8e549b79a350bca53c454,
     "rpc/core_rpc_server_commands_defs.h",
     45..=55
 )]
-#[allow(clippy::needless_pass_by_value)] // TODO: remove after impl
-fn compress_integer_array(array: Vec<u64>) -> error::Result<Vec<u64>> {
-    todo!()
+#[cfg(feature = "epee")]
+fn compress_integer_array(array: &[u64]) -> error::Result<Vec<u8>> {
+    let capacity = array.len() * (u64::BITS as usize * 8 / 7 + 1);
+    let mut vec = Vec::<u8>::with_capacity(capacity);
+
+    for i in array {
+        write_varint(*i, &mut vec)?;
+    }
+
+    Ok(vec)
 }
 
-/// Used for [`Distribution::CompressedBinary::compressed_data`].
-///
-/// TODO: for handler code. This should already be provided in the field.
+/// Used for [`Distribution::CompressedBinary::distribution`].
 #[doc = crate::macros::monero_definition_link!(
     cc73fe71162d564ffda8e549b79a350bca53c454,
     "rpc/core_rpc_server_commands_defs.h",
     57..=72
 )]
-#[allow(clippy::needless_pass_by_value)] // TODO: remove after impl
-fn decompress_integer_array(array: Vec<u64>) -> Vec<u64> {
-    todo!()
+fn decompress_integer_array(mut array: &[u8]) -> Vec<u64> {
+    let capacity = array.len();
+    let mut vec = Vec::<u64>::with_capacity(capacity);
+
+    while let Ok(varint) = read_varint(&mut array) {
+        vec.push(varint);
+    }
+
+    vec
```

Comment:
(De)compression functions.

Ref: https://github.com/monero-project/monero/blob/cc73fe71162d564ffda8e549b79a350bca53c454/src/rpc/core_rpc_server_commands_defs.h#L45-L72.

Reply:
Monero has 2 varint formats, the one we are using is the epee varint, monerod seems to be using the consensus serialisation varint, which is defined in `monero-serai`: https://github.com/serai-dex/serai/blob/6f34c2ff77ef8a9f8839dbf63f35dcd6dc33f4b3/networks/monero/io/src/lib.rs#L60

Reply:
might need to leave this as a todo until #201

Reply:
Changed bodies to `todo!()` for now.